components/Tabs/src/private/MDCItemBarCell.m

```diff
@@ -33,8 +33,9 @@
 /// Padding between top of the cell and the badge.
 static const CGFloat kBadgeTopPadding = 6;
 
-/// Maximum width of a badge. This allows for 3 characters before truncation.
-static const CGFloat kBadgeMaxWidth = 22;
+/// Maximum badge text character length. Badge text longer than this number of characters will
```

Comment:
Tip: To be clear, there is no limit to the number of characters Bottom Navigation badges support.  Here we are imposing an artificial limit of 4. I still recommend limiting the label's frame rather than attempting to count rendered glyphs.  This seems like a lot of extra effort compared to calculating the frame.